### PR TITLE
Add error state support to DaxTextInput

### PR DESCRIPTION
**Pull request by CrisBarreiro**
Task/Issue URL: https://app.asana.com/0/488551667048375/1204338910522509/f 

### Description
Add error state support to DaxtextInput and use it in NetP waitlist

### Steps to test this PR

_Check error status_
- [ ] Go to NetP through settings
- [ ] Select I have an Invite Code
- [ ] Type a random bunch of characters
- [ ] Press the arrow/ok button on the keyboard
- [ ] Check that it submits the code and triggers the error state
- [ ] Check error state styling is correct

_Check single line and caps_
- [ ] Go to NetP through settings
- [ ] Select I have an Invite Code
- [ ] Type a random bunch of characters and check the field is single line
- [ ] Check the keyboard is launched with caps on

### CODE REVIEW

**Comment by CrisBarreiro** [comment]
* `develop`
  * **#4120** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/4120?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @CrisBarreiro and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: common/common-ui/src/main/res/values/attrs-text-input.xml [hunk 1]
```diff
@@ -19,6 +19,7 @@
         <attr name="android:hint" />
         <attr name="android:text" />
         <attr name="android:minLines" />
+        <attr name="capitalizeKeyboard" format="boolean" />
```
Comment (malmstein) [inline common/common-ui/src/main/res/values/attrs-text-input.xml:22]: nit: capitalize keyboard or text? the parameter for TextView is `textAllCaps`, we could reuse it if you wanted.

**Review by malmstein — APPROVED** [review]
Thanks @CrisBarreiro! Could you add the new `capitalizeKeyboard` parameter to https://app.asana.com/0/0/1203532494940541/f?

File: common/common-ui/src/main/res/values/attrs-text-input.xml (hunk 1 above)
Comment (CrisBarreiro) [reply common/common-ui/src/main/res/values/attrs-text-input.xml:22]: The original implementation was capitalizeKeyboard, which is slightly different. In this case, the keyboard is displayed in all caps by default, but the user could still switch to lowercase (or use an external keyboard) and text wouldn't be capitalized automatically. I did it this way to mimic the original behavior as much as possible.

**Pull request merged**